packages/repository-modernizer/src/restructure-config.js

```diff
@@ -338,5 +340,152 @@ async function deleteEmptyConfigDir(oldConfigFilePath) {
+                            key,
+                            val,
+                            newConfigFilePath
+                        );
+                        contentToBeWritten.push(str + ",");
+                    }
+                }
+                if (contentToBeWritten.length > 1) {
+                    let lastStr = contentToBeWritten[
+                        contentToBeWritten.length - 1
+                    ].substring(
+                        0,
+                        contentToBeWritten[contentToBeWritten.length - 1]
+                            .length - 1
+                    );
+                    contentToBeWritten.pop();
+                    contentToBeWritten.push(lastStr);
+                }
+            }
+            contentToBeWritten.push("}");
+            await util.writeDataToFileAsync(path, contentToBeWritten);
+        } else {
+            if (newConfigFilePath.endsWith(".config")) {
+                path =
+                    newConfigFilePath.substring(
+                        0,
+                        newConfigFilePath.lastIndexOf(".config")
+                    ) + ".cfg.json";
+            } else {
+                path =
+                    newConfigFilePath.substring(
+                        0,
+                        newConfigFilePath.lastIndexOf(".cfg")
+                    ) + ".cfg.json";
+            }
+            contentToBeWritten.push("{");
+            fileContent = fileContent.filter((a) => a);
+            for (let line = 0; line < fileContent.length; line++) {
+                let configLine = fileContent[line].trim();
+                let key = configLine.substring(0, configLine.indexOf("="));
+                let val = configLine.substring(configLine.indexOf("=") + 1);
+                let str = removeUnwantedChars(key, val, newConfigFilePath);
+                if (key != "") {
+                    if (line == fileContent.length - 1) {
+                        contentToBeWritten.push(str);
+                    } else {
+                        contentToBeWritten.push(str + ",");
+                    }
+                }
+            }
+            contentToBeWritten.push("}");
+            await util.writeDataToFileAsync(path, contentToBeWritten);
+        }
+        fs.unlinkSync(newConfigFilePath);
+    }
+}
+/**
+ * @param String key which will be added as key in resultant file
+ * @param String val which will be added as value in resultant file
+ * @param String FilePath the new config file path in ui.config package
+ *
+ * format generate line which will be added to resultant cfg.json file
+ */
+function removeUnwantedChars(key, val, filePath) {
+    val = val.replace(/\\/g, "");
+    let str = "";
+    if (filePath.endsWith(".xml")) {
+        if (val.charAt(0) == "{") {
+            let type = val.substring(1, val.indexOf("}"));
+            val = val.substring(val.indexOf("}") + 1);
+            val = val.replace(/[^\x20-\x7E]/gim, "");
```

Comment:
Can you add a comment for every regex as to what is the purpose. It will help in future debugging.